### DISCUSSION TITLE
Chrome共用字典壓縮

### DISCUSSION THREAD

**Discussion by Eileen050824** [1101-GS4719-C/Python-Community-GS4719-C#1927] [General]
最近看到Chrome實現共用字典壓縮技術，可使經常更新程式碼的網站提高壓縮比，更能實現90％的壓縮率，對我來說是蠻深奧也有點新鮮。
*壓縮：在輸入的資料中尋找重複的序列，並利用該資訊創建一個更小的輸出。
![image](https://github.com/NCU-GS4719-Python/Python-Community/assets/160737635/610bee86-0708-4237-9993-7a1bdf35f526)
在稍微了解和資料查詢後，單看功能及其介紹層面，我覺得壓縮技術在網路應用的效果很好，主要是可以大幅減少資源載入時間。瀏覽器常用的Brotli和ZStandard演算法，搭配針對特定資源模式的自定義字典一起使用，特別是內容高度相似的字典，都能夠使用壓縮字典技術提高壓縮率，我相信這對系統處理效率和框架更新都會有所助益。
但同時我也在其他文章看到，使用字典壓縮資源，過去瀏覽器解壓縮時採用一種SDCH的流程，但是實作上存在安全性挑戰。Chrome方雖說新版壓縮已經解決了這些安全性問題，但對我來說這存在一定的疑慮，雖然我不確定那個安全性問題所引發的後續效應可能多深(網路沒有特別說到)，但畢竟該方沒有提供什麼透明的實驗證明，也沒有什麼測試者或使用者去支持這點...
身為這個領域的小白，我只單就這件事做正反兩面的看法(我相信任何事物都有其優缺點)並些許分享，如果有理解錯誤或了解這領域的大大有什麼意見都歡迎指教！
來源：iThome

**Comment by wanderer657** [comment]
我認為Chrome的共用字典壓縮技術能極大提升網頁載入速度，對於更新頻繁的網站尤其有效。然而，安全隱患也不可忽視，可能影響用戶隱私和數據安全。因此，我們需在提高性能的同時，注重安全，確保用戶數據安全。
